Sources/MarathonCore/Export.swift

```diff
@@ -0,0 +1,165 @@
+        }
+    }
+}
+
+// MARK: - Task
+
+internal final class ExportTask: Task, Executable {
+    private typealias Error = ExportError
+
+    func execute() throws {
+
+        guard let exportScriptPath = arguments.first?.asScriptPath() else {
+            throw Error.missingPath
+        }
+
+        let script = try scriptManager.script(atPath: exportScriptPath, allowRemote: false)
+
+        let destination = try destinationFolder()
+        try overwriteIfNeeded(destination: destination, scriptName: script.name)
+
+        let tempFile = try script.folder.createFile(named: script.name.asScriptPath())
+        try tempFile.write(data: try script.folder.file(named: "OriginalFile").read())
+
+        try perform(export(tempFile, to: destination), orThrow: Error.failedToExportScript(tempFile.name))
+        try tempFile.delete()
+    }
+
+    private func destinationFolder() throws -> Folder {
+        let folder: Folder
+        if arguments.count > 1 && !arguments[1].contains("--force") {
+            folder = try FileSystem().createFolderIfNeeded(at: arguments[1])
+        } else {
+            folder = FileSystem().currentFolder
+        }
+        return folder
+    }
+
+    private func overwriteIfNeeded(destination: Folder, scriptName: String) throws {
+        let exportPath = destination.path.appending(scriptName)
+        let existingProjectFolder = try? destination.subfolder(atPath: scriptName)
+
+        if existingProjectFolder != nil && !arguments.contains("--force") {
+            printer.output("⚠️  A directory already exists at \(exportPath)")
+            printer.output("❓  Are you sure you want to overwrite it? (Type 'Y' to confirm)")
+            if readLine()?.lowercased() == "n" {
+                exit(1)
+            }
+        }
+        try existingProjectFolder?.delete()
+    }
+
+    private func export(_ file: File, to folder: Folder) throws {
+        let projectFolder = try folder.createSubfolder(named: file.scriptName)
+
+        let packageFile = try projectFolder.createFile(named: "Package.swift")
+        let packages = try resolvePackages(from: file)
+        let packageFileString = try makePackageFileString(for: file, with: packages)
+        try packageFile.write(string: packageFileString)
+
+        let sourcesFolder = try projectFolder.createSubfolder(named: "Sources")
+        let scriptFileString = try makeFileStringWithInlineDependencies(for: file, using: packages)
+        let scriptFile = try sourcesFolder.createFile(named: file.name)
+        try scriptFile.write(string: scriptFileString)
+    }
+
+    private func resolvePackages(from file: File) throws -> [Package] {
+        let importNames = try file.importNames()
+        let allManagedPackages = packageManager.addedPackages
+        let scriptPackages: [Package] = importNames.flatMap { name in
+            if let scriptPackage = allManagedPackages.first(where: { $0.name.lowercased() == name.lowercased() }) {
```

Comment:
I think this code would be a lot simpler (and more efficient) if you instead would `filter` `allManagedPackages` based on whether `importNames` includes the package name, like this:

```swift
let importNames = try Set(file.importNames())
let scriptPackages = packageManager.addedPackages.filter { importNames.contains($0.name) }
```

Do we need to compare against lowercase versions of the names?